cache/dns_cache.ml

```diff
@@ -22,11 +22,11 @@ let compare_rank a b = match a, b with
   | ZoneTransfer, _ -> 1
   | _, ZoneTransfer -> -1
   | AuthoritativeAnswer signed, AuthoritativeAnswer signed' ->
-    Bool.compare signed signed'
+    Bool.compare (Option.is_some signed) (Option.is_some signed')
   | AuthoritativeAnswer _, _ -> 1
   | _, AuthoritativeAnswer _ -> -1
   | AuthoritativeAuthority signed, AuthoritativeAuthority signed' ->
-    Bool.compare signed signed'
+    Bool.compare (Option.is_some signed) (Option.is_some signed')
```

Comment:
both of these compare may need revisiting -- taking into consideration what we would like to keep in the map if we have different values (certainly the one with rrsig is better)

Reply:
Perhaps one strategy could be to pick one with a later expiry